Source/Core/Core/AchievementManager.h

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#ifdef USE_RETRO_ACHIEVEMENTS
+#include <mutex>
+#include <string>
+#include <thread>
+
+#include <rcheevos/include/rc_api_user.h>
+#include <rcheevos/include/rc_runtime.h>
+
+#include "Common/Event.h"
+
+class AchievementManager
+{
+public:
+  static AchievementManager* GetInstance();
+  void Init();
+  void Login(const std::string& password);
+  bool IsLoggedIn();
+  void Logout();
+  void Shutdown();
+
+private:
+  AchievementManager() = default;
+
+  void ThreadProc();
+
+  void VerifyCredentials(const std::string& password);
+
+  void DestroyCredentials();
+
+  template <typename RcRequest, typename RcResponse>
+  static void Request(RcRequest rc_request, RcResponse* rc_response,
+                      std::function<int(rc_api_request_t*, const RcRequest*)> init_request,
```

Comment:
`std::function` needs `#include <functional>`

Reply:
both `std::function` instances should also be passed by const reference, since they're executed immediately in the function